watsong/static/js/webapp.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * getCenter returns the center of an HTML element in x-y coordinates.
+ *
+ * @param {HTMLElement} elem element to find the center of
+ * @return {{x: number, y: number}} center of elem in as object with .x and .y
+ */
+function getCenter(elem) {
+  const { top, left, width } = elem.getBoundingClientRect();
+  return { y: top + width / 2, x: left + width / 2 };
+}
+
+/**
+ *  getCoord returns the position of a touch or click event
+ *
+ * @param {MouseEvent | TouchEvent} event touch/mouse event
+ * @return {{x: number, y: number}} position of touch or click with .x and .y
+ */
+function getCoord(event) {
+  if (event instanceof MouseEvent) {
+    return { x: event.clientX, y: event.clientY };
+  } else {
+    // return coordinates of first touch
+    const firstTouch = event.touches[0];
+    return { x: firstTouch.clientX, y: firstTouch.clientY };
+  }
+}
+
+
+/**
+ * initializeDial makes the DOM element twistable and sends HTTP request when the twisting stops.
+ *
+ * @param {HTMLElement} elem is the `div` element that contains the image.
+ * @param {string} dialName is the dial name to be sent to the backend
+ */
+function initializeDial(elem, dialName) {
+  var currentAngle = 0, prevAngle = 0, initialAngle = 0, center = getCenter(elem);
+
+  elem.onmousedown = onTwistStart;
+  elem.ontouchstart = onTwistStart;
+
+  /**
+   * onTwistStart is called when a user touches or clicks down on `elem`
+   *
+   * @param {MouseEvent | TouchEvent} event touch/mouse event
+   */
+  function onTwistStart(event) {
+    event.preventDefault();
+
+    center = getCenter(elem);
+    const { x, y } = getCoord(event);
+
+    initialAngle = Math.atan2(x - center.x, y - center.y);
+
+    document.onmouseup = onStopTwisting;
+    document.onmousemove = onTwist;
+
+    document.ontouchend = onStopTwisting;
+    document.ontouchmove = onTwist;
+  }
+
+  /**
+   * onTwist is called when a mouse moves while clicked, or dragged via touch.
+   *
+   * @param {MouseEvent | TouchEvent} event touch/mouse event
+   */
+  function onTwist(event) {
+    event.preventDefault();
+
+    const { x, y } = getCoord(event);
+
+    const newAngle = Math.atan2(x - center.x, y - center.y);
+
+    currentAngle = -(prevAngle + newAngle - initialAngle);
+
+    $(elem).css("transform", `rotate(${currentAngle}rad)`);
+  }
+
+
+  /**
+   * onStopTwisting is called when a user lets go of the dial (`elem`)
+   */
+  function onStopTwisting() {
+    // stop twisting when mouse button is released
+    document.onmouseup = null;
+    document.onmousemove = null;
+
+    document.ontouchend = null;
+    document.ontouchmove = null;
+
+    prevAngle = -currentAngle;
```

Comment:
You don't use this variable later

Reply:
It's a variable with scope for the entire function, it's used again in `onTwist`.